### DISCUSSION TITLE
How to correctly setup meow so it works nicely with my original key bindings?

### DISCUSSION THREAD

**Discussion by drcxd** [meow-edit/meow#301] [Q&A] [answered]
Hi, I am just switching to meow from evil and found it is really powerful and simple to use once I getting familiar with its style of thinking.

However, there is are a few questions bothering me:

1. Let's say in vanilla emacs `C-x b` is bound to `switch-to-buffer` and `C-x C-b` is bound to `ibuffer`. It is fair to say that in vanilla emacs `switch-to-buffer` is bound to an easier to use key binding than `ibuffer`. However, after using mewo, I need to press `SPC x SPC b` to execute `switch-to-buffer`, but only `SPC x b` to execute `ibuffer`. This makes `switch-to-buffer` the harder to use. I see this as a contradiction with the default design of the vanilla emacs key bindings. How could I make `SPC x b` bind to `switch-to-buffer` and `SPC x SPC b` bing to `ibuffer` in a non-intrusive way?

I figured out that I can add code to my meow settings like this
```
(global-set-key (kbd "C-x C-b") `switch-to-buffer)
(global-set-key (kbd "C-x b) `ibuffer)
```

However, this makes me unable to use the vanilla key-bindings, which cannot be called non-intrusive, from my perspective.

2. This is about minor-mode specific key bindings. For example, I am using lsp-mode and I want to add some key bindings in the `lsp-mode-map`. I have code like
```
;; global key bindings
(global-set-key (kbd "C-c ll") `lsp)
;; mode-specific key bindings
(define-key lsp-mode-map (kbd "C-c lf") 'lsp-find-definition)
;; more mode-specific key bindings...
```
I can press `SPC ll` to enable lsp-mode, but I can not execute `lsp-find-definition` by pressing `SPC lf` even if `lsp-mode` is already loaded. Instead, I have to use `SPC c SPC l f` which is somewhat cumbersome.

I have seen that I could define a leader key like this
```
(meow-leader-define-key '("t" . "C-c l"))
```
to use `SPC t` as prefix instead of `SPC c SPC l`, but this will override any key bindings prefixed by `C-c t`. To me, it's like to use some mode-specific key bindings easily, they have to take up the space of two keymaps. Is there any workaround for this problem?

3. When using evil, I have some key bindings prefixed by `SPC c`. After switching to meow, I found that it is natural to re-prefix those keys to `C-c c`. However, when using KEYPAD mode, I have to type `SPC c SPC c` to access those key bindings, which is inconvenient. The only solution I found is to re-prefix them to`C-c *` where `*` is something other than "x/c/h/m/g". However, if I do not use meow but vanilla emacs, then there is no reason I should do that. This is also not non-intrusive. How could resolve this confliction?

**Comment by DogLooksGood** [comment, marked as answer]
1. Just use something like `SPC b`. According to the key binding convention, `C-c <letter>` is preserved for user custom key bindings. So by default, it's empty. When you define a binding in `C-c <letter>` space, you got two ways to execute. `C-c <letter>` and `SPC <letter>`. 
2. I guess the reason is you have something on C-c C-l`?

**Reply by drcxd** [reply]
Thanks for your answer, for the second question, `C-c C-l` is bound by c++-mode-map. My point is, while the global key binding `C-c ll` can be accessed by `SPC l` in KEYPAD mode, why key bindings in `lsp-mode-map` also prefixed by `C-c l` cannot be accessed by `SPC l`.
Also, do you mind share some insight about my third question?

**Reply by DogLooksGood** [reply]
Sorry for the late response, was quite busy these days. 

For the third question, there's no good solution. The leader feature is used to solve some tricky key bindings. It's recommended to do some necessary key binding instead of heavy usages. Since C is occupied by `C-c`, there's no way to do `C-c c` in a simpler way. 

I think this is what you need for mixed sub-keymap:
 `(setq meow-keypad-leader-dispatch "C-c")`
